### ISSUE TITLE
Do we want to 'get' hardware id from OS rather then generate

### ISSUE THREAD

**Issue by andreievg** [openmsupply/open-msupply#131]
**Conclusion** we **SHOULD** query hardware id form os. For android we'll have to pass it form native (we already do this for folder paths)

Just a follow up from https://github.com/openmsupply/open-msupply/issues/129, I've asked @roxy-dao to just generate hardware id and store it in app data. Rather than getting hardware id from OS.

The rational for this:

* If we are having a multiplatform app, it is a good idea to try and minimise the differences in implementations
* Although there are cross platform crates for getting hardware id, there doesn't seem to be uniform one that works for android also
* As per above two, just generating hardware id and storing it in app data should be sufficient (also it's one logic for all platforms)

Unintended consequence would be: If you clear app data, hardware id is reset, you'll have to reset it on central server. This is rare case, and it's super easy to reset hardware id now with synchronisation UI we have.

So i would vote for generating it, any objections ?

**Comment by mark-prins** [comment]
I disagree - the h/w id we're using for mobile is annoying.. though not annoying enough to require an immediate fix. I see that  https://github.com/openmsupply/mobile/issues/1290 is still open

Mobile is following a similar process to what you're suggesting, and has the issue that a reinstall or clearing of app data requires a config update to the sync settings on the server. I thought that we'd be able to use a device specific ID instead, which could be retained even through a reinstall of the app.

> minimise the differences in implementations

Which is kinda the same point as the next one. Or do you mean difference in h/w id format? Either way, don't see it is a huge issue

> cross platform crates for getting hardware id

The only outlier here is android isn't it? I thought that there were crates that would work on linux, windows & mac? I don't think that's too bad 

Basically, I don't think the minor difference of approach for android is worth the trade off of needing to update the server whenever you reset a device.

**Comment by andreievg** [comment]
I am not very strong about this whole thing. I think we would need to be on Android 8 + to get HW ID. (just asked Nick he said mostly 8 +).

Just asked @Gwalior-84 one more time, he would like to see hardware id being queried from OS.

Although I am still not sure why we would be resetting app data on frequent basis, I would assume that during fail to initialise we are able to clear data on failure (without full data reset, since that's not available on desktop), so I think that issue might be mitigated by that feature ?

**Comment by andreievg** [comment]
Ok more from @Gwalior-84, to pretty much tilt it for me toward queries from OS

`Andrei, [16/06/2022 at 10:45:58 AM]:`
One more question, if we have functionality to clear database without doing factory reset or clearing app data, does it mitigate the issue with generated UUID ?

`Mark Glover, [16/06/2022 at 11:18:23 AM]:`
The factory resetting is rarely (never?) done due to any issue with mSupply.  It is normally done because something has gone wrong with MEMDM.

After trying less drastic measures like just turning the device on / off, trying different internet connection, etc. Factory reset is the way of 'starting again from scratch'.

Since v5.7, [HW reset](https://docs.msupply.org.nz/synchronisation:sync_sites#hardware_id) can now be done by any user with permission, and doesn't need to be done by a TMF support worker.  It is still a hassle to have to reset it, but it is not as much of a hassle as it used to be.  Even so, I would argue that the HW ID should persist through factory resets.
